August RC (#6775)
microsoft/vscode-python: CHANGELOG.md

# Changelog

> ## 2019.7.0-rc (30 July 2019)

Reviewer: ```suggestion
## 2019.8.0-rc (30 July 2019)
```

1. Clarify regexes used for decreasing indentation.
   ([#6333](https://github.com/Microsoft/vscode-python/issues/6333))
1. Add new plot viewer button images and fix button colors in different themes.
   ([#6336](https://github.com/Microsoft/vscode-python/issues/6336))
1. Update telemetry property name for Jedi memory usage.
   ([#6339](https://github.com/Microsoft/vscode-python/issues/6339))
1. Fix png scaling on non standard DPI. Add 'enablePlotViewer' setting to allow user to render pngs instead of svg files.
   ([#6344](https://github.com/Microsoft/vscode-python/issues/6344))
1. Do best effort to download the experiments and use it in the very first session only
   ([#6348](https://github.com/Microsoft/vscode-python/issues/6348))
1. Linux can pick the wrong kernel to use when starting the interactive window.
   ([#6375](https://github.com/Microsoft/vscode-python/issues/6375))
1. Add missing keys for data science interactive window button tooltips in package.nls.json
   ([#6386](https://github.com/Microsoft/vscode-python/issues/6386))
1. Fix overwriting of cwd in the path list when discovering tests.
   ([#6417](https://github.com/Microsoft/vscode-python/issues/6417))

> 1. Fixes a bug in pytest test discovery.  Patch by Rainer Dreyer.

Reviewer: ```suggestion
1. Fixes a bug in pytest test discovery. 
    (thanks Rainer Dreyer)
```

   ([#6463](https://github.com/Microsoft/vscode-python/issues/6463))
1. Fix debugging to work on restarting the jupyter kernel.
   ([#6502](https://github.com/Microsoft/vscode-python/issues/6502))
1. Escape key in the interactive window moves to the delete button when auto complete is open. Escape should only move when no autocomplete is open.
   ([#6507](https://github.com/Microsoft/vscode-python/issues/6507))
1. Render plots as png, but save an svg for exporting/image viewing. Speeds up plot rendering.
   ([#6526](https://github.com/Microsoft/vscode-python/issues/6526))
1. Import get_ipython at the start of each imported jupyter notebook if there are line magics in the file
   ([#6574](https://github.com/Microsoft/vscode-python/issues/6574))

> 1. Fix problem where we retrieved and rendered old codelenses for multiple imports of jupyter notebooks if cells in the resultant import file were executed without saving the file to disk

Reviewer: ```suggestion
1. Fix a problem where we retrieved and rendered old codelenses for multiple imports of jupyter notebooks if cells in the resultant import file were executed without saving the file to disk.
```

   ([#6582](https://github.com/Microsoft/vscode-python/issues/6582))

> 1. Ptvsd install for jupyter debugging should check version without actually importing into the jupyter kernel

Reviewer: ```suggestion
1. PTVSD install for jupyter debugging should check version without actually importing into the jupyter kernel.
```

   ([#6592](https://github.com/Microsoft/vscode-python/issues/6592))
1. Fix pandas version parsing to handle strings.
   ([#6595](https://github.com/Microsoft/vscode-python/issues/6595))

> 1. Unpin the version of ptvsd in the install and add -U

Reviewer: ```suggestion
1. Unpin the version of ptvsd in the install and add `-U`.
```

   ([#6718](https://github.com/Microsoft/vscode-python/issues/6718))
1. Fix stepping when more than one blank line at the end of a cell.
   ([#6719](https://github.com/Microsoft/vscode-python/issues/6719))
1. Render plots as png, but save an svg for exporting/image viewing. Speeds up plot rendering.
   ([#6724](https://github.com/Microsoft/vscode-python/issues/6724))
1. Fix random occurrences of output not concatenating correctly in the interactive window.
   ([#6728](https://github.com/Microsoft/vscode-python/issues/6728))
1. In order to debug without '#%%' defined in a file, support a Debug Entire File.
   ([#6730](https://github.com/Microsoft/vscode-python/issues/6730))

> 1. Add support for Run Below back.

Reviewer: ```suggestion
1. Add support for "Run Below" back.
```

1. Remove Debug Cell and Run Cell from the command palette. They should both be 'Debug Current Cell' and 'Run Current Cell'
   ([#6754](https://github.com/Microsoft/vscode-python/issues/6754))

### Code Health

1. Log processes executed behind the scenes in the extension output panel.
   ([#1131](https://github.com/Microsoft/vscode-python/issues/1131))
1. Specify `pyramid.scripts.pserve` when creating a debug configuration for Pyramid
   apps instead of trying to calculate the location of the `pserve` command.
   ([#2427](https://github.com/Microsoft/vscode-python/issues/2427))
1. UI Tests using [selenium](https://selenium-python.readthedocs.io/index.html) & [behave](https://behave.readthedocs.io/en/latest/).
   ([#4692](https://github.com/Microsoft/vscode-python/issues/4692))
1. Upload coverage reports to [coveralls](https://coveralls.io/github/microsoft/vscode-python).
   ([#5999](https://github.com/Microsoft/vscode-python/issues/5999))
1. Upgrade Jedi to version 0.13.3.
   ([#6013](https://github.com/Microsoft/vscode-python/issues/6013))
1. Add unit tests for `client/activation/serviceRegistry.ts`.
   ([#6163](https://github.com/Microsoft/vscode-python/issues/6163))

> 1. Remove test.ipynb from root folder.

Reviewer: ```suggestion
1. Remove `test.ipynb` from the root folder.
```

   ([#6212](https://github.com/Microsoft/vscode-python/issues/6212))
1. Fail the `smoke tests` CI job when the smoke tests fail.
   ([#6253](https://github.com/Microsoft/vscode-python/issues/6253))
1. Add a bunch of perf measurements to telemetry.
   ([#6283](https://github.com/Microsoft/vscode-python/issues/6283))
1. Retry failing debugger test (retry due to intermittent issues on `Azure Pipelines`).
   ([#6322](https://github.com/Microsoft/vscode-python/issues/6322))
1. Update version of `isort` to `4.3.21`.
   ([#6369](https://github.com/Microsoft/vscode-python/issues/6369))
1. Functional test for debugging jupyter cells.
   ([#6377](https://github.com/Microsoft/vscode-python/issues/6377))
1. Consolidate telemetry.
   ([#6451](https://github.com/Microsoft/vscode-python/issues/6451))
1. Removed npm package `vscode`, and added to use `vscode-test` and `@types/vscode` (see [here](https://code.visualstudio.com/updates/v1_36#_splitting-vscode-package-into-typesvscode-and-vscodetest) for more info).
   ([#6456](https://github.com/Microsoft/vscode-python/issues/6456))

> 1. Fix variable explorer exclude test to be less strict

Reviewer: ```suggestion
1. Fix the variable explorer exclude test to be less strict.
```

   ([#6525](https://github.com/Microsoft/vscode-python/issues/6525))
1. Merge ArgumentsHelper unit tests into one file.
   ([#6583](https://github.com/Microsoft/vscode-python/issues/6583))

> 1. Fix jupyter remote tests to respect new notebook 6.0 output format

Reviewer: ```suggestion
1. Fix jupyter remote tests to respect new notebook 6.0 output format.
```

   ([#6625](https://github.com/Microsoft/vscode-python/issues/6625))
1. Unit Tests for DataScience Error Handler.
   ([#6670](https://github.com/Microsoft/vscode-python/issues/6670))

> 1. Fix DataExplorer tests after accessibility fixes

Reviewer: ```suggestion
1. Fix DataExplorer tests after accessibility fixes.
```

   ([#6711](https://github.com/Microsoft/vscode-python/issues/6711))

> 1. Bump version of [PTVSD](https://pypi.org/project/ptvsd/) to 4.3.0.

Reviewer: Should include the relevant highlights from https://github.com/microsoft/ptvsd/releases/tag/v4.3.0 (this would also be where to mention we know about the erroneous terminal output).
